x-pack/auditbeat/processors/sessionmd/add_session_metadata.go

```diff
@@ -53,25 +56,34 @@ type addSessionMetadata struct {
 	providerType string
 }
 
+func gen_registry(reg *monitoring.Registry, base string) *monitoring.Registry {
+	// if more than one instance of the DB is running, start to increment the metrics keys.
+	// This is kind of an edge case, but best to handle it so monitoring does not explode
+	id := 0
+	if reg.GetRegistry(base) != nil {
+		id = int(instanceID.Add(1))
+	}
+
+	regName := base
```

Comment:
The old code always did `instanceID.Add()` and then checked, so it started at 1, your code starts at 0 for the first one.

Is this what you want? I think it should be
```
id = instanceId.Add(1)
reg.GetRegistry(base)
if id > 1
   regName = sprintf...
else
    regName = base
...
```

Also the function name is not camelCased, any special reason?

Reply:
Yeah, I've been doing a lot of rust so my brain just fails to go into go mode sometimes :p

Reply:
The problem is that we call `genRegistry` multiple times during init to set up the metrics, so that the counter needs to be tied to a metrics name. I don't really like this solution either, and it only exists to handle the edge case of someone starting up multiple identical processors. A longer-term solution would be to make the monitoring code less panic-happy.

Reply:
hmm but wasn't the old code handling this as well? just starting on 0 ?

Reply:
but what about changing the starting id? I still don't get what was wrong before that it had to change.

Reply:
I don't think the actual end behavior has changed? If we have more than one metrics registry with a given name, add a digit to the end of the name. This is mostly so the registry doesn't panic if someone configures the processor twice.